indigo_api/templatetags/indigo.py

```diff
@@ -21,59 +25,27 @@ def work_resolver_url(context, work):
     return context.get('resolver_url', settings.RESOLVER_URL) + frbr_uri
 
 
-def make_beautiful(provisions, commenceable_provisions):
-    runs = []
-    current_run = []
-    previous = None
-
-    def add_to_current(element):
-        current_run.append({'type': element.type, 'num': element.num.strip('.')})
-
-    def end_current():
-        if len(current_run) < 3:
-            run = [f'{item["type"]} {item["num"]}' for item in current_run]
-        else:
-            run = [f'{current_run[0]["type"]} {current_run[0]["num"]}–{current_run[-1]["num"]}']
-        runs.append(run)
-        return []
-
-    for p in commenceable_provisions:
-        if p.id in provisions:
-            if not previous:
-                add_to_current(p)
-                previous = p
-                continue
-
-            if previous.id in provisions and previous.type != p.type:
-                # current run is not safe to add to (different type)
-                current_run = end_current()
-            add_to_current(p)
-
-        elif previous and previous.id in provisions:
-            current_run = end_current()
-
-        previous = p
-
-    # we've finished checking all commenceable provisions; add the last group
-    # (but not if it's empty)
-    if current_run:
-        end_current()
-
-    return ', '.join(', '.join(p) for p in runs)
-
-
 @register.simple_tag
 def commenced_provisions_description(document, commencement, uncommenced=False):
-    if uncommenced:
-        provisions = [p.id for p in document.uncommenced_provisions()]
-    else:
-        provisions = commencement.provisions
+    # To get the provisions relevant to each row of the table,
+    # use the earlier of the document's expression date and the relevant commencement's date
+    date = document.expression_date
+    if commencement and date > commencement.date:
+        date = commencement.date
 
-    commenceable_provisions = document.commenceable_provisions()
+    provisions = deepcopy(document.work.all_commenceable_provisions(date))
+    provision_ids = document.work.all_uncommenced_provision_ids(document.expression_date) if uncommenced else commencement.provisions
 
-    return make_beautiful(provisions, commenceable_provisions)
+    beautifier = CommencementsBeautifier(commenced=not uncommenced)
+    # decorate the ToC with useful information
+    provisions = beautifier.decorate_provisions(provisions, provision_ids)
+    return beautifier.make_beautiful(provisions)
 
 
 @register.simple_tag
 def commencements_relevant_at_date(document):
```

Comment:
I guess this doesn't need to be a template tag since we can call this directly on the document object. Maybe in the past we passed in a date...

Reply:
I raise this simply because I'm trying to understand the semantic methods that we have on the Document object, and `commencements_relevant_at_expression_date` is really pretty decent.